### PR TITLE
NAS-109782 / 21.04 / Run DHCP in parallel

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

File: src/middlewared/middlewared/plugins/network.py [hunk 1]
```diff
@@ -1925,13 +1925,13 @@ async def sync(self, wait_dhcp=False):
...
-                await self.sync_interface(interface, wait_dhcp, sync_interface_opts[interface])
-            except Exception:
-                self.logger.error('Failed to configure {}'.format(interface), exc_info=True)
+        vlans, non_vlans = bisect(lambda x: x.startswith('vlan'), filter(lambda i: not i.startswith('br'), interfaces))
+        if non_vlans:
+            await asyncio.wait([self._sync_interface_wrapper(interface, wait_dhcp, sync_interface_opts[interface])
+                                for interface in non_vlans])
+        if vlans:
+            await asyncio.wait([self._sync_interface_wrapper(interface, wait_dhcp, sync_interface_opts[interface])
+                                for interface in vlans])
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/network.py:1934]: Wont we remain with the same problem here? waiting dhcp on an unconfigured interface before we set up bridges?

Comment (themylogin) [reply src/middlewared/middlewared/plugins/network.py:1934]: @william-gr all this code is executed only for configured interfaces. The interface with no carrier was explicitly configured for DHCP.

Comment (william-gr) [reply src/middlewared/middlewared/plugins/network.py:1934]: Same question applies

Comment (william-gr) [reply src/middlewared/middlewared/plugins/network.py:1934]: If a DHCP times out for an interface that is not related to bridge, bridge will only be configured after timeout of that unrelated interface, no?

**Review by william-gr — APPROVED** [review]

**Pull request merged**